### PR TITLE
Add Ubuntu Jammy CI

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Part of https://github.com/ignition-tooling/release-tools/issues/566

## Summary

Enable CI for Ubuntu 22.04 (Jammy)

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by scpeters** [comment]
this needs libignition-common4-dev debs on jammy

**Comment by scpeters** [comment]
`pxr` fails to build on Ubuntu 22.04 (jammy). Perhaps we should limit `pxr` testing to 20.04 (focal) for now?

cc @adlarkin @ahcorde

**Comment by adlarkin** [comment]
> `pxr` fails to build on Ubuntu 22.04 (jammy). Perhaps we should limit `pxr` testing to 20.04 (focal) for now?

That sounds good to me :+1:

**Comment by scpeters** [comment]
> > `pxr` fails to build on Ubuntu 22.04 (jammy). Perhaps we should limit `pxr` testing to 20.04 (focal) for now?
> 
> That sounds good to me 👍

done in https://github.com/ignitionrobotics/sdformat/pull/864/commits/990d548f0b251b1f2ccbc8facb4f11cd43b1c388

File: .github/ci/before_cmake.sh [hunk 1]
```diff
@@ -7,12 +7,16 @@ BUILD_DIR=`pwd`
 cd /tmp
 
 # check that we can compile USD from sources (only Focal)
+return_code=0
+if [ "$(lsb_release -r -s)" != "20.04" ]; then
+  return_code=$(($return_code + 1))
```
Comment (adlarkin) [inline .github/ci/before_cmake.sh:12]: since `return_code` is initialized to `0` above, can this be simplified to `return_code=1`?

Comment (scpeters) [reply .github/ci/before_cmake.sh:12]: I think it's easier to add new code and move things around in its current form.

**Comment by adlarkin** [comment]
> `pxr` fails to build on Ubuntu 22.04 (jammy). Perhaps we should limit `pxr` testing to 20.04 (focal) for now?

Looking at this more closely, I believe it could be because you need to set certain environment variables to build `pxr` from source. Here are the environment variables set on Focal: https://github.com/ignitionrobotics/sdformat/blob/2a517d1bc563a831c68f9456897659ed054243e6/.github/workflows/ci.yml#L21-L23

@scpeters have you tried setting these environment variables on Jammy to see if it fixes the Jammy build?

**Comment by scpeters** [comment]
> > `pxr` fails to build on Ubuntu 22.04 (jammy). Perhaps we should limit `pxr` testing to 20.04 (focal) for now?
> 
> Looking at this more closely, I believe it could be because you need to set certain environment variables to build `pxr` from source. Here are the environment variables set on Focal:
> 
> https://github.com/ignitionrobotics/sdformat/blob/2a517d1bc563a831c68f9456897659ed054243e6/.github/workflows/ci.yml#L21-L23
> 
> @scpeters have you tried setting these environment variables on Jammy to see if it fixes the Jammy build?

I had not tried this, but I just did in https://github.com/ignitionrobotics/sdformat/commit/bd0ab93e0d73da4a142b62b65c4c48c8196f448e

it is still failing:

~~~
2022-03-07T23:36:40.2577991Z [ 60%] Building CXX object pxr/base/work/CMakeFiles/work.dir/singularTask.cpp.o
2022-03-07T23:36:40.3566396Z In file included from /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.cpp:24:
2022-03-07T23:36:40.3567151Z /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.h: In lambda function:
2022-03-07T23:36:40.3568067Z /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.h:111:21: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
2022-03-07T23:36:40.3568421Z   111 |                     size_t old = count;
2022-03-07T23:36:40.3568644Z       |                     ^~~~~~
2022-03-07T23:36:40.3568845Z       |                     std::size_t
2022-03-07T23:36:40.3569371Z In file included from /usr/include/c++/11/bits/atomic_base.h:35,
2022-03-07T23:36:40.3569771Z                  from /usr/include/c++/11/atomic:41,
2022-03-07T23:36:40.3570258Z                  from /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.h:31,
2022-03-07T23:36:40.3570810Z                  from /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.cpp:24:
2022-03-07T23:36:40.3571440Z /usr/include/x86_64-linux-gnu/c++/11/bits/c++config.h:280:33: note: 'std::size_t' declared here
2022-03-07T23:36:40.3571824Z   280 |   typedef __SIZE_TYPE__         size_t;
2022-03-07T23:36:40.3572133Z       |                                 ^~~~~~
2022-03-07T23:36:40.3585857Z In file included from /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.cpp:24:
2022-03-07T23:36:40.3586408Z /tmp/usd_binaries/USD-21.11/pxr/base/work/singularTask.h:113:56: error: 'old' was not declared in this scope
2022-03-07T23:36:40.3586778Z   113 |                         !count.compare_exchange_strong(old, 0));
2022-03-07T23:36:40.3587036Z       |                                                        ^~~
2022-03-07T23:36:40.3630030Z make[2]: *** [pxr/base/work/CMakeFiles/work.dir/build.make:177: pxr/base/work/CMakeFiles/work.dir/singularTask.cpp.o] Error 1
~~~

I noticed tbb deprecation warnings, so I'm wondering if it's related to https://github.com/PixarAnimationStudios/USD/issues/1471

**Comment by adlarkin** [comment]
> I had not tried this, but I just did in [bd0ab93](https://github.com/ignitionrobotics/sdformat/commit/bd0ab93e0d73da4a142b62b65c4c48c8196f448e)
> 
> it is still failing:
>
> ...
> 
> I noticed tbb deprecation warnings, so I'm wondering if it's related to [PixarAnimationStudios/USD#1471](https://github.com/PixarAnimationStudios/USD/issues/1471)

Okay, thanks for checking. Since there seem to be upstream issues on Jammy, I think we can stick with the approach of disabling `pxr` on Jammy for now. Would you mind leaving a comment after [this line](https://github.com/ignitionrobotics/sdformat/blob/2a517d1bc563a831c68f9456897659ed054243e6/.github/ci/before_cmake.sh#L9) that references the upstream issues on Jammy? That way, we can re-visit this later on and try to re-enable `pxr` on Jammy whenever things are fixed upstream.

**Comment by scpeters** [comment]
> > I had not tried this, but I just did in [bd0ab93](https://github.com/ignitionrobotics/sdformat/commit/bd0ab93e0d73da4a142b62b65c4c48c8196f448e)
> > it is still failing:
> > ...
> > I noticed tbb deprecation warnings, so I'm wondering if it's related to [PixarAnimationStudios/USD#1471](https://github.com/PixarAnimationStudios/USD/issues/1471)
> 
> Okay, thanks for checking. Since there seem to be upstream issues on Jammy, I think we can stick with the approach of disabling `pxr` on Jammy for now. Would you mind leaving a comment after [this line](https://github.com/ignitionrobotics/sdformat/blob/2a517d1bc563a831c68f9456897659ed054243e6/.github/ci/before_cmake.sh#L9) that references the upstream issues on Jammy? That way, we can re-visit this later on and try to re-enable `pxr` on Jammy whenever things are fixed upstream.

I opened #869 to track this and added a comment in https://github.com/ignitionrobotics/sdformat/pull/864/commits/ed6220245f72c432097de2194872b235079b2460

**Review by adlarkin — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**